### ISSUE TITLE
Upgrade to 3.2 breaks GridFS

### ISSUE THREAD

**Issue by malixsys** [devconcept/multer-gridfs-storage#14]
While 3.1 works perfectly, 3.2.x for me gets:

```bash
TypeError: this.db.on is not a function
    at GridFSStorage._setDb (/Users/malix/Temp/nz/node_modules/multer-gridfs-storage/lib/gridfs.js:253:6)
    at MongoClient.connect (/Users/malix/Temp/nz/node_modules/multer-gridfs-storage/lib/gridfs.js:209:12)
    at connectCallback (/Users/malix/Temp/nz/node_modules/mongodb/lib/mongo_client.js:515:5)
    at /Users/malix/Temp/nz/node_modules/mongodb/lib/mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
``` 
connectiing with: 
```typescript
import * as GridFsStorage from 'multer-gridfs-storage';
const storage = new GridFsStorage({
    url: config.secrets.uris,
    //...
});```

!?

**Comment by devconcept** [comment]
The `3.2` update added support for mongoose. All other changes are minor. Please provide more details about the error. Are you using typescript?

**Comment by Yaant** [comment]
I'm having the same issue:

> TypeError: this.db.on is not a function
> and so on

with combination of mongodb@2.2 and multer-gridfs-storage@3.2. 

Switching to multer-gridfs-storage@3.1 solves the problem.

See attached zip with a code demonstrating the issue. Unpack and run:

> npm i
> node .

[test.zip](https://github.com/devconcept/multer-gridfs-storage/files/2412444/test.zip)

Should be irrelevant, but just in case: I'm using NodeJS 8.9.0 x64 on Windows 7.

**Comment by devconcept** [comment]
Fixed in version `3.2.3`

For TS users: Typings for v3 (caching and other features) are already updated and will be available in a few days.

**Comment by devconcept** [comment]
Typings [updated](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29498)
